### PR TITLE
fix(docs): warn java users setting alloc/lock to 0 is not a good idea

### PR DESCRIPTION
**Pull request by korniltsev**
Add the following sentence

```
Setting to 0 registers every event, causing significant overhead and results in large profiles, making it not ideal for production. We recommend a starting value of 512k, adjusting as needed.
```

Maybe it will stop users setting alloc=0

https://github.com/grafana/pyroscope/issues/2341

### CODE REVIEW

File: docs/sources/configure-client/language-sdks/java.md [hunk 1]
```diff
@@ -116,26 +116,25 @@ for configuration in multiple sources: system properties, environment variables,
...
+|---------------------------------------------|-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| `PYROSCOPE_FORMAT`                          | sets the profiler output format. The default is `collapsed`, but in order to support multiple formats it must be set to `jfr`.                                                                                                                                                                                                                                                                              |
```
Comment (petethepig) [inline docs/sources/configure-client/language-sdks/java.md:121]: side note: we should probably switch to jfr by default

File: docs/sources/configure-client/language-sdks/java.md [hunk 2]
```diff
@@ -116,26 +116,25 @@ for configuration in multiple sources: system properties, environment variables,
...
+| `PYROSCOPE_PROFILER_EVENT`                  | sets the profiler event. With JFR format enabled, this event refers to one of the possible CPU profiling events: `itimer`, `cpu`, `wall`. The default is `itimer`.                                                                                                                                                                                                                                          |
+| `PYROSCOPE_PROFILER_ALLOC`                  | sets the allocation threshold to register the events, in bytes (equivalent to `--alloc=` in `async-profiler`). The default value is `""` - empty string, which means that allocation profiling is disabled. Setting to 0 registers every event, causing significant overhead and results in large profiles, making it not ideal for production. We recommend a starting value of 512k, adjusting as needed. |
```
Comment (petethepig) [inline docs/sources/configure-client/language-sdks/java.md:123]:
```suggestion
| `PYROSCOPE_PROFILER_ALLOC`                  | sets the allocation threshold to register the events, in bytes (equivalent to `--alloc=` in `async-profiler`). The default value is `""` - empty string, which means that allocation profiling is disabled. Setting it to 0 will register every event, causing significant CPU and network overhead, making it not suitable production environments. We recommend setting a starting value of 512k and adjusting it as needed. |
```

**Review by petethepig — APPROVED** [review]
made it sound a bit better

File: docs/sources/configure-client/language-sdks/java.md (hunk 1 above)
Comment (korniltsev) [reply docs/sources/configure-client/language-sdks/java.md:121]: hehe, I already did it this morning https://github.com/grafana/pyroscope-java/pull/112/files#diff-99f8fe330b16fd778c4e54482fe3f5e9ba425b5dd458fbcf9858e3ef74e0d786R70

**Pull request merged**